src/handle.rs

```diff
@@ -0,0 +1,36 @@
+#![allow(dead_code)]
+
+use std::ptr::NonNull;
+
+/// MUST NOT IMPLEMENT COPY or CLONE
+/// This struct OWN the pointer
+/// and have a F implementation that will call F to free the pointer.
```

Comment:
Rewrite comment to be more formal and clearer. I suggest.
```
`Handle` MUST NOT implement `Copy` or `Clone` because it assumes it's the sole owner of the pointer and it will free it upon `Drop`ing.
```

Reply:
Oh and it should be a normal comment (i.e. `//`) and not a documentation comment (i.e. `///`).